File: services/main-service/match-service/src/main/java/com/kjeldsen/match/state/BallState.java
package com.kjeldsen.match.state;

import com.kjeldsen.match.entities.Player;
import com.kjeldsen.player.domain.PitchArea;
import lombok.Value;

@Value
public class BallState {

    /*
     * Represents the position of the ball and which player is in control of it.
     */
    Player player;

    PitchArea area;

    BallHeight height;

    public static BallState init() {
        return new BallState(null, null, null);
    }
}

File: frontends/beta-frontend/shared/models/match/BallState.ts
import { Player } from '../player/Player';
import { PitchArea } from './PitchArea';

export interface BallState {
  height: BallHeight;
  player: Player;
  pitchArea: PitchArea;
}

export enum BallHeight {
  GROUND = 'GROUND',
  LOW = 'LOW',
  HIGH = 'HIGH',
}
